chore: update xml reader kb (#2010)
telerik/reporting-docs: knowledge-base/xml-support.md

---
title: Does Telerik Reporting support XML as a DataSource?
description: "Learn about whether Telerik Reporting supports using XML data in reports out-of-the-box, and what alternatives are available in case it is not officially supported."
type: troubleshooting

> page_title: Does Telerik Reporting support XML as a DataSource?

Reviewer: ```suggestion
page_title: How to Use XML as a DataSource in Telerik Reporting
```

    <Book>
        <BookID>1</BookID>
        <Title>The Great Gatsby</Title>
        <Author>F. Scott Fitzgerald</Author>
        <Year>1925</Year>
        <Price>12.99</Price>
    </Book>
    <Book>
        <BookID>2</BookID>
        <Title>1984</Title>
        <Author>George Orwell</Author>
        <Year>1949</Year>
        <Price>14.99</Price>
    </Book>
    <Book>
        <BookID>3</BookID>
        <Title>To Kill a Mockingbird</Title>
        <Author>Harper Lee</Author>
        <Year>1960</Year>
        <Price>13.99</Price>
    </Book>
    <Book>
        <BookID>4</BookID>
        <Title>Pride and Prejudice</Title>
        <Author>Jane Austen</Author>
        <Year>1813</Year>
        <Price>11.99</Price>
    </Book>
</BooksDataSet>
````

The XML data above can be integrated in the report by programmatically reading it and creating a [DataSet](https://learn.microsoft.com/en-us/dotnet/api/system.data.dataset) object from it, which can be used via the ObjectDataSource Component]({%slug telerikreporting/designing-reports/connecting-to-data/data-source-components/objectdatasource-component/overview%}). To implement such a solution, you may follow the steps below:

> 1. (Recommended) Create a *typed DataSet*(in other wods -  `.xsd` file) from your XML. This is recommended to do because, otherwise, all of the fields in the data will be treated as **strings**, and conversion functions would need to be used in the report(s). For example, the following `Books.xsd` file can be created for the XML data above:

Reviewer: ```suggestion
1. (Recommended) Create a *typed DataSet*(in other words -  `.xsd` file) from your XML. This is recommended to do because, otherwise, all of the fields in the data will be treated as **strings**, and conversion functions would need to be used in the report(s). For example, the following `Books.xsd` file can be created for the XML data above:
```
